### PR TITLE
Porting more 010 inet forms/types

### PR DESCRIPTION
**Pull request by vertexmc**

### CODE REVIEW

File: synapse/models/inet.py [hunk 1]
```diff
@@ -234,6 +237,51 @@ def _normPyStr(self, valu):
         except Exception as e:
             raise s_exc.BadTypeValu(valu)
 
+class Rfc2822Addr(s_types.Type):
```
Comment (invisig0th) [inline synapse/models/inet.py]: Maybe s_types.Str so you get a bunch of the below for free?

Comment (vertexmc) [reply synapse/models/inet.py]: What would we get for free?

**Comment by vertexmc** [comment]
This is ready for review

File: synapse/models/inet.py (hunk 1 above)
Comment (invisig0th) [reply synapse/models/inet.py]: Go ahead and leave it as-is for now, but plz add an indxcmpr for ^= to do prefix matching :)

Comment (vertexmc) [reply synapse/models/inet.py]: sure

Comment (vertexmc) [reply synapse/models/inet.py]: added

**Pull request merged**